app/services/idv/analytics_events_enhancer.rb

```diff
@@ -1,112 +1,91 @@
-      idv_in_person_usps_proofing_enrollment_code_email_received
-      idv_in_person_usps_proofing_results_job_completed
-      idv_in_person_usps_proofing_results_job_deadline_passed_email_exception
-      idv_in_person_usps_proofing_results_job_deadline_passed_email_initiated
-      idv_in_person_usps_proofing_results_job_email_initiated
-      idv_in_person_usps_proofing_results_job_enrollment_incomplete
-      idv_in_person_usps_proofing_results_job_enrollment_updated
-      idv_in_person_usps_proofing_results_job_exception
-      idv_in_person_usps_proofing_results_job_please_call_email_initiated
-      idv_in_person_usps_proofing_results_job_started
-      idv_in_person_usps_proofing_results_job_unexpected_response
-      idv_in_person_usps_proofing_results_job_user_sent_to_fraud_review
-      idv_in_person_usps_request_enroll_exception
-      idv_ipp_deactivated_for_never_visiting_post_office
-      idv_link_sent_capture_doc_polling_complete
-      idv_link_sent_capture_doc_polling_started
-      idv_mail_only_warning_visited
-      idv_native_camera_forced
-      idv_not_verified_visited
-      idv_phone_use_different
-      idv_request_letter_visited
-      idv_sdk_selfie_image_capture_closed_without_photo
-      idv_sdk_selfie_image_capture_failed
-      idv_sdk_selfie_image_capture_opened
-      idv_selfie_image_added
-      idv_session_error_visited
-      idv_threatmetrix_response_body
-      idv_usps_auth_token_refresh_job_completed
-      idv_usps_auth_token_refresh_job_network_error
-      idv_usps_auth_token_refresh_job_started
-      idv_verify_by_mail_enter_code_submitted
-      idv_verify_by_mail_enter_code_visited
-      idv_verify_in_person_troubleshooting_option_clicked
-      idv_warning_action_triggered
-      idv_warning_shown
-    ].to_set.freeze
+    EXCLUDED_FRONTEND_EVENT_METHODS = [
+      :idv_acuant_sdk_loaded,
+      :idv_barcode_warning_continue_clicked,
+      :idv_barcode_warning_retake_photos_clicked,
+      :idv_camera_info_error,
+      :idv_camera_info_logged,
+      :idv_capture_troubleshooting_dismissed,
+      :idv_consent_checkbox_toggled,
+      :idv_image_capture_failed,
+      :idv_in_person_location_submitted,
+      :idv_in_person_ready_to_verify_sp_link_clicked,
+      :idv_in_person_ready_to_verify_what_to_bring_link_clicked,
+      :idv_link_sent_capture_doc_polling_complete,
+      :idv_link_sent_capture_doc_polling_started,
+      :idv_native_camera_forced,
+      :idv_sdk_error_before_init,
+      :idv_sdk_selfie_image_capture_closed_without_photo,
+      :idv_sdk_selfie_image_capture_failed,
+      :idv_sdk_selfie_image_capture_initialized,
+      :idv_sdk_selfie_image_capture_opened,
+      :idv_sdk_selfie_image_re_taken,
+      :idv_sdk_selfie_image_taken,
+      :idv_selfie_image_added,
+      :idv_verify_in_person_troubleshooting_option_clicked,
+      :idv_warning_action_triggered,
+      :idv_warning_shown,
+    ].freeze
+
+    EXCLUDED_JOB_EVENT_METHODS = [
```

Comment:
Is there a way we could just detect that idv_session is not available / present and not attempt to enhance in that case? That way we don't have to maintain a list of events that fire in the worker

Reply:
That's what I would love to move to but felt like that might be beyond the scope of the ticket. Open to moving back to draft and rethinking that though!

Reply:
I really like this idea, but I also agree that it's a deviation from what the story says.

Reply:
Committed to writing a spike ticket to figure out how to move away from this design.